element/src/commonMain/kotlin/RoundedRectangleElement.kt

```diff
@@ -0,0 +1,42 @@
+package com.juul.krayon.element
+
+import com.juul.krayon.kanvas.Kanvas
+import com.juul.krayon.kanvas.Paint
+
+public class RoundedRectangleElement : Element() {
```

Comment:
In the DOM world, are there specific elements for "rounded rectangle"? Or, would you just style a basic DOM element?

Reply:
Rounded rectangles are a little complicated, and definitely one of the areas where different platforms deviate the most.

The `rect` element in SVG supports `rx` and `ry` for corner roundness. These elements always effect all four corners, where `rx` determines the width of the arc and `ry` determines the height of the arc. This is also the model used for Android's `Canvas`.

Interestingly, this is not the same model used by HTML/CSS _or_ by Android's `RoundRectShape`, which use a model more similar to what I have here (specified by corner, but each corner is uniform/cannot be wider than it is tall).

The HTML `Canvas` and associated JS has no concept of a rounded rectangle at all.

This doesn't answer your question directly, but I think that's because there's not really a good answer there. In a real D3 world, I'd probably try and solve this problem with CSS if possible instead of making it part of the DOM.

Reply:
For now, I'd like to keep a separate element type for rounded rectangles, since the basic rectangle is so fundamental and there's a pretty meaningful speed difference in rendering the two due to path creation.

Reply:
Makes sense. Thanks for the thorough explanation!